src/modules/accounting/accounting.py

```diff
@@ -327,3 +301,29 @@ def _is_bunker(self, blockstamp: ReferenceBlockStamp) -> bool:
         )
         logger.info({'msg': 'Calculate bunker mode.', 'value': bunker_mode})
         return bunker_mode
+
+    @lru_cache(maxsize=1)
+    def get_extra_data(self, blockstamp: ReferenceBlockStamp) -> ExtraData:
+        consensus_version = self.w3.lido_contracts.accounting_oracle.get_consensus_version(blockstamp.block_hash)
+
+        chain_config = self.get_chain_config(blockstamp)
+        stuck_validators = self.lido_validator_state_service.get_lido_newly_stuck_validators(blockstamp, chain_config)
+        logger.info({'msg': 'Calculate stuck validators.', 'value': stuck_validators})
+        exited_validators = self.lido_validator_state_service.get_lido_newly_exited_validators(blockstamp)
+        logger.info({'msg': 'Calculate exited validators.', 'value': exited_validators})
+        orl = self.w3.lido_contracts.oracle_report_sanity_checker.get_oracle_report_limits(blockstamp.block_hash)
+
+        if consensus_version == 1:
+            return ExtraDataService.collect(
+                stuck_validators,
+                exited_validators,
+                orl.max_items_per_extra_data_transaction,
+                orl.max_node_operators_per_extra_data_item,
+            )
+
+        return ExtraDataServiceV2.collect(
+            stuck_validators,
+            exited_validators,
+            orl.max_items_per_extra_data_transaction,
+            orl.max_node_operators_per_extra_data_item,
+        )
```

Comment:
Yeah, this approach is much better if we want to support all Oracle versions.
And is better if there were more than one versioned condition.
But since there is only one version checking condition and a lot of refactorings, let's go with current version.
But I'll keep this in head and maybe use this approach for TE